### PR TITLE
Add Reporter to TranslatorSyncerExtensionParams

### PR DESCRIPTION
**Pull request by ashleywang1**
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/2719

### CODE REVIEW

File: changelog/v1.4.0-beta1/translator-syncer-extension-reporter.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+changelog:
+  - type: NEW_FEATURE
```
Comment (kdorosh) [inline changelog/v1.4.0-beta1/translator-syncer-extension-reporter.yaml:2]: perhaps `NON_USER_FACING`, there are no user-facing changes until we leverage this in Gloo-Enterprise and add the new error reporting there

File: changelog/v1.4.0-beta1/translator-syncer-extension-reporter.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+    description: Allow TranslatorSyncerExtension's to be able to report errors and warnings to the CRDs too.
```
Comment (kdorosh) [inline changelog/v1.4.0-beta1/translator-syncer-extension-reporter.yaml:3]: nit,
```suggestion
    description: Allow `TranslatorSyncerExtension`s to be able to report errors and warnings to Gloo CRs too.
```

File: changelog/v1.4.0-beta1/translator-syncer-extension-reporter.yaml (hunk 1 above)
Comment (kdorosh) [reply changelog/v1.4.0-beta1/translator-syncer-extension-reporter.yaml:2]:
```suggestion
  - type: NON_USER_FACING
```

**Comment by ashleywang1** [comment]
/kick "gateway tests with virtual service" flake

**Comment by ashleywang1** [comment]
/kick "updates Envoy endpoints even if proxy is rejected " test flake

**Pull request merged**